src/amo/components/LandingPage.js

```diff
@@ -108,9 +115,22 @@ export class LandingPageBase extends React.Component {
   }
 }
 
-export function mapStateToProps(state, ownProps) {
+export function mapStateToProps(
+  state, ownProps, _apiAddonType = apiAddonType, _log = log
+) {
+  let addonType;
+  try {
+    addonType = _apiAddonType(ownProps.params.visibleAddonType);
+  } catch (err) {
+    if (err.message.match('not found in API_ADDON_TYPES_MAPPING')) {
```

Comment:
How hard would it be to throw and catch a custom error like `AddonTypeNotFound` ? I'm not sure if we've done custom error classes yet. If not, you can use [es6-error](https://github.com/bjyoungblood/es6-error) to make `instanceof` work.